src/modules/sway/bar.cpp

```diff
@@ -68,6 +119,9 @@ void BarIpcClient::onIpcEvent(const struct Ipc::ipc_response& res) {
     if (payload.isMember("visible_by_modifier")) {
       // visibility change for hidden bar
       signal_visible_(payload["visible_by_modifier"].asBool());
+    } else if (payload.isMember("change")) {
+      spdlog::trace("BarIpcClient::onIpcEvent: got event {}", payload["change"].asString());
+      signal_last_modifier_(false);
```

Comment:
What is the event type here?

Reply:
I used this to store that the last modifier key press was followed by an event which was the "reason" for the key press, and not clear the visible status for the bar.

Reply:
My reason to use a signal here was that I assumed that one `onIpcEvent` could be active while another one could be started, but not sure how `ipc_.signal_event.connect(...)` works.

Reply:
> not sure how ipc_.signal_event.connect(...) works

The logic is following:
1. everything connected to the `ipc_` signals directly will be executed in the ipc worker thread, sequentially (in order of receiving the events from the socket). But each ipc connection will have its own worker thread.
2. everything connected to SafeSignal instances will be queued for execution in the main (UI) thread, in the order of event emission.
3. SafeSignal.emit called from the main (UI) thread will bypass the queue and get processed synchronously.

tl;dr: event handlers in different threads may run in parallel, events in the same thread are serialized.